e2e/framework/addon/vault.go

```diff
@@ -187,13 +190,39 @@ func (l *Vault) initVault() error {
 	l.KubernetesAuthPath = "mykubernetes"              // see configure-vault.sh
 	l.KubernetesAuthRole = "external-secrets-operator" // see configure-vault.sh
 
-	ginkgo.By("Creating vault TLS secret")
-	err = l.chart.config.CRClient.Create(context.Background(), sec)
+	// vault-config contains vault init config and policies
+	files, err := os.ReadDir(fmt.Sprintf("%s/vault-config", AssetDir()))
+	if err != nil {
+		return err
+	}
+	sec := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "vault-tls-config",
+			Namespace: l.Namespace,
+		},
+		Data: map[string][]byte{},
+	}
+	_, err = controllerutil.CreateOrUpdate(context.Background(), l.chart.config.CRClient, sec, func() error {
```

Comment:
I really think this client should be context aware. using context.Background all over here is dangerous because it cannot be cancelled. this can cause test leakage